model/task/db.go

```diff
@@ -815,42 +820,37 @@ func FindUniqueBuildVariantNamesByTask(projectId string, taskName string) ([]str
 
 	group := bson.M{
 		"$group": bson.M{
-			"_id":            taskName,
+			"_id":            bson.M{BuildVariantDisplayNameKey: "$" + BuildVariantDisplayNameKey},
 			buildVariantsKey: bson.M{"$addToSet": "$" + BuildVariantKey},
 		},
 	}
-	unwindAndSort := []bson.M{
-		{
-			"$unwind": "$build_variants",
-		},
-		{
-			"$sort": bson.M{
-				"build_variants": 1,
-			},
+
+	project := bson.M{
```

Comment:
Are we sure we don't want to still unwind? It's possible for build variants to change display names; if multiple build variants match a display name, is it okay that we're only keeping the first one?